### PR TITLE
Add multiplayer compatibility using MPAPI

### LINKED ISSUE #146: Compatibility with multiplayer mod.
Any thoughts on making this mod compatible with RimWorld's multiplayer mod. As of now it is marked as not working.

> Setting a Bill to “Prisoners Only” Desyncs. It also breaks the initialization of the game at random even without the use of the mp mod.

### PR DESCRIPTION
**Pull request by Thomas107500**
1. changed Unity Random methods to using Verse Random 
2. moved some code to separate methods for multiplayer compatibility
3. added MPAPI nuget package

Closes #146

### CODE REVIEW

File: Source/PrisonLabor.csproj [hunk 1]
```diff
@@ -37,40 +39,44 @@
...
     <Reference Include="System.Core" />
     <Reference Include="System.Xml.Linq" />
     <Reference Include="System.Data.DataSetExtensions" />
     <Reference Include="System.Data" />
     <Reference Include="System.Xml" />
     <Reference Include="UnityEngine.CoreModule">
-      <HintPath Condition="Exists('../../_RimWorldData/Managed/')">../../_RimWorldData/Managed/UnityEngine.CoreModule.dll</HintPath>
-      <HintPath Condition="Exists('..\..\..\Gry\Steam\')">..\..\..\Gry\Steam\steamapps\common\RimWorld\RimWorldWin64_Data\Managed\UnityEngine.CoreModule.dll</HintPath>
-      <Private>False</Private>
+      <HintPath>..\..\..\..\..\..\Program Files (x86)\Steam\steamapps\common\RimWorld\RimWorldWin64_Data\Managed\UnityEngine.CoreModule.dll</HintPath>
```
Comment (Hazzer) [inline Source/PrisonLabor.csproj:56]: Please revert this changes. As resolving reference please set path in visual project settings. Something like that:
obraz

Nuget part is ok.

**Review by Hazzer — CHANGES_REQUESTED** [review]

File: Source/PrisonLabor.csproj (hunk 1 above)
Comment (Thomas107500) [reply Source/PrisonLabor.csproj]: Done, i think i fixed it

File: Source/CompatibilityPatches/Multiplayer.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using UnityEngine;
+using Verse;
+
+namespace PrisonLabor.CompatibilityPatches
+{
+    [StaticConstructorOnStartup]
+    public static class Multiplayer
```
Comment (Hazzer) [inline Source/CompatibilityPatches/Multiplayer.cs:18]: Just wonder. This approach requires to always release mod with Multiplayer.dll or have that one on active mod list.

Do you think it will work if you do something like here: https://github.com/Aviuz/PrisonLabor/blob/master/Source/CompatibilityPatches/Quarry.cs as check if MP is enabled and just included in project references?

Comment (Thomas107500) [reply Source/CompatibilityPatches/Multiplayer.cs:18]: not sure what you mean by Multiplayer.dll, do you mean the actual Multiplayer mod itself? because you don't need to include the mod here or in the active mod list, only the MPAPI dll is required

**Comment by Hazzer** [comment]
Just noticed: Multiplayer mod is on 1.2 right now. This pull request will work only with prison labor played at 1.3 Rimworld version.

**Comment by Thomas107500** [comment]
Oh for that actually Multiplayer mod for 1.3 is now on beta right now in the Multiplayer mod discord, lots of people tried it and it is pretty stable right now, as for this patch at least 5 people tested it and no one reported there is any issue so far

**Comment by Thomas107500** [comment]
For your reference, i think i should include the link to the mod's discord group and documentation of MPAPI

[MPAPI documentation](https://hackmd.io/@rimworldmultiplayer/docs/https%3A%2F%2Fhackmd.io%2F%40rimworldmultiplayer%2Fdev-mapi)

 [Multiplayer mod discord](https://discord.gg/S4bxXpv)

**Review by Hazzer — APPROVED** [review]

**Comment by Hazzer** [comment]
> 
> 
> For your reference, i think i should include the link to the mod's discord group and documentation of MPAPI
> 
> [MPAPI documentation](https://github.com/Parexy/Multiplayer/wiki/API)
> 
> [Multiplayer mod discord](https://discord.gg/S4bxXpv)

Yup, I thought that MPAPI is that same dll as mod files. It is clear and looks good now. Thanks for contribution.
